### PR TITLE
Revision history dnd

### LINKED ISSUE #1965: Release UI: Promote from history within channels (same arch)

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #1965
- Adds possibility for dragging revision from history panel to promote it
- Refactors d'n'd by extracting shared functionality and styles

### QA
- ./run or  https://snapcraft-io-canonical-web-and-design-pr-2052.run.demo.haus/
- go to Releases page of any snap
- click on revision cell to open history panel
- drag any revisions in history to promote it to some channel
- make sure dragging revision cells and channel rows still work

Screenshot 2019-07-01 at 15 33 29

### CODE REVIEW

**Comment by therealjuan** [comment]
If you open up the contextual menu on a channel and drag another channel, perhaps the menu should close automatically?

**Comment by bartaz** [comment]
@therealjuan Good catch, thanks. There are some issues with menus/tooltips not hiding on rows/cells that are not active in dnd (because or devmode revisions or other reasons). And because they are not using drag'n'drop they are unaware of it themselves (so can't hide).
To fix that we would need some global state, so that whole UI knows drag is in progress and can react to that.

Let's have it fixed separately, as it's not related to history at all.

**Comment by therealjuan** [comment]
Another one to discuss about revision history:

image
Should the current revision `14239` have white background whereas the other revisions should have a background color like #f7f7f7 and on mouse over become white? Basically working like the channels in other parts of the interface.

Something like this:

image

File: static/js/publisher/release/components/revisionsList.js [hunk 1]
```diff
@@ -252,6 +252,7 @@ class RevisionsList extends Component {
         <table className="p-revisions-list">
           <thead>
             <tr>
+              <th width="30px" />
```
Comment (Lukewh) [inline static/js/publisher/release/components/revisionsList.js]: _looks the other way_

Comment (bartaz) [reply static/js/publisher/release/components/revisionsList.js]: If only there was a better way to do it without unnecessary CSS overhead ¯\_(ツ)_/¯

**Comment by bartaz** [comment]
Added separate issue #2053 to cover highlighting active revision in history mentioned by @therealjuan

**Pull request merged**